src/Zynga/Framework/StorableObject/Collections/Map/V2/Base.hh

```diff
@@ -0,0 +1,188 @@
+<?hh // strict
+
+namespace Zynga\Framework\StorableObject\Collections\Map\V2;
+
+use
+  Zynga\Framework\StorableObject\Collections\Map\V2\Exporter\Base as Exporter
+;
+use
+  Zynga\Framework\StorableObject\Collections\Map\V2\Importer\Box as BoxImporter
+;
+use
+  Zynga\Framework\StorableObject\Collections\Map\V2\Importer\Storable as StorableImporter
+;
+use
+  Zynga\Framework\StorableObject\Collections\V2\Interfaces\StorableMapCollection
+;
+use
+  Zynga\Framework\StorableObject\V1\Exceptions\OperationNotSupportedException
+;
+use Zynga\Framework\StorableObject\V1\Fields;
+use Zynga\Framework\StorableObject\V1\Fields\Generic as FieldsGeneric;
+use Zynga\Framework\StorableObject\V1\Interfaces\ExportInterface;
+use Zynga\Framework\StorableObject\V1\Interfaces\FieldsInterface;
+use Zynga\Framework\StorableObject\V1\Interfaces\ImportInterface;
+use Zynga\Framework\StorableObject\V1\Interfaces\StorableObjectInterface;
+use Zynga\Framework\Type\V1\Interfaces\TypeInterface;
+
+class Base<Tv> implements StorableMapCollection<Tv> {
+  /**
+   * Native Map only supports int keys or string keys
+   * https://docs.hhvm.com/hack/reference/class/HH.Map/
+   * We are forcing it to support only string as for data with
+   * int keys we can make use of StorableVector
+   */
+  private Map<string, Tv> $map;
```

Comment:
For generics in Hacklang, the type parameter is something like 'T<k,v>'. Which, personally, I am not a fan off because it's not descriptive. What do you think of changing the type parameter to something more descriptive like 'Map<TypeValue>'. It would fall in line more with the DotNet standards of more descriptive names.